src/lib/networkCode.js

```diff
@@ -0,0 +1,49 @@
+import { hex2patp } from 'urbit-ob';
+
+import jssha256 from 'js-sha256';
+
+function shax(buf) {
+  const hashed = jssha256.sha256.array(buf);
+  return Buffer.from(hashed);
+}
+
+function hexToArrayBuffer(hex) {
+  return Buffer.from(hex, 'hex').reverse();
+}
+
+function buf2hex(buf) {
+  return Buffer.from(buf)
+    .reverse()
+    .toString('hex');
+}
+
+function shas(buf, salt) {
+  return shax(xor(salt, shax(buf)));
+}
+
+function xor(a, b) {
```

Comment:
Do we not have a xor implementation in one of the crypto things we import?

Reply:
There's one in a transitive dependency, should I make it a direct dep and use that?

Reply:
Hmm, I think that's still best? Re-implementing primitives like this feels like trouble. Let's do it.

Reply:
In a sick turn of events the dependency is buggier than the handrolled version. I could bring it in and hack around it, but it might just be better to leave it

Reply:
Agreed. At the first sign of "even more trouble", run!

Reply:
Ready to go then!